src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/ProtobufSerializerHelper.cs

```diff
@@ -0,0 +1,233 @@
+        WriteRawByte(buffer, ref cursor, (byte)value);
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static void WriteBoolWithTag(byte[] buffer, ref int cursor, int fieldNumber, bool value)
+    {
+        WriteTag(buffer, ref cursor, fieldNumber, WireFormat.WireType.Varint);
+        WriteRawByte(buffer, ref cursor, value ? (byte)1 : (byte)0);
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static void WriteFixed64WithTag(byte[] buffer, ref int cursor, int fieldNumber, ulong value)
+    {
+        WriteTag(buffer, ref cursor, fieldNumber, WireFormat.WireType.Fixed64);
+        WriteRawLittleEndian64(buffer, ref cursor, value);
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static void WriteInt64WithTag(byte[] buffer, ref int cursor, int fieldNumber, ulong value)
+    {
+        WriteTag(buffer, ref cursor, fieldNumber, WireFormat.WireType.Varint);
+        WriteRawVarint64(buffer, ref cursor, value);
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static void WriteDoubleWithTag(byte[] buffer, ref int cursor, int fieldNumber, double value)
+    {
+        WriteTag(buffer, ref cursor, fieldNumber, WireFormat.WireType.Fixed64);
+        WriteRawLittleEndian64(buffer, ref cursor, (ulong)BitConverter.DoubleToInt64Bits(value));
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static void WriteLengthCustom(byte[] buffer, ref int cursor, int length)
+    {
+        WriteRawVarintCustom(buffer, ref cursor, (uint)length);
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static void WriteLength(byte[] buffer, ref int cursor, int length)
+    {
+        WriteRawVarint32(buffer, ref cursor, (uint)length);
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static void WriteRawVarintCustom(byte[] buffer, ref int cursor, uint value)
+    {
+        int index = 0;
+
+        // Loop until all 7 bits from the integer value have been encoded
+        while (value > 0)
+        {
+            byte chunk = (byte)(value & 0x7F); // Extract the least significant 7 bits
+            value >>= 7; // Right shift the value by 7 bits to process the next chunk
+
+            // If there are more bits to encode, set the most significant bit to 1
+            if (index < 3)
+            {
+                chunk |= 0x80;
+            }
+
+            buffer[cursor++] = chunk; // Store the encoded chunk
+            index++;
+        }
+
+        // If fewer than 3 bytes were used, pad with zeros
+        while (index < 2)
+        {
+            buffer[cursor++] = 0x80;
+            index++;
+        }
+
+        while (index < 3)
+        {
+            buffer[cursor++] = 0x00;
+            index++;
+        }
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static void WriteRawVarint32(byte[] buffer, ref int cursor, uint value)
```

Comment:
I'm concerned about copying only the methods from Protobuf. The best approach would be to obtain the file from Protobuf and then move your custom implementations to a different file. Before copying the content, have you verified the license of the Protobuf repository? Here are a few approaches to address this issue

1. Review the license information and collaborate with the maintainer on how to obtain the code from another repository.
1. Customize and develop our own solution.

Reply:
@rajkumar-rangaraj - I have not copied the methods as is. The implementations may look same but that's because of the encoding.

> Customize and develop our own solution.

I think that is what I am doing. Is there something specific you see that could be an issue?

Reply:
It's only a couple of methods that have an exact match, and I agree that, apart from those, it's all your implementation. If you're customizing, It would be better to change the method names instead of following the naming conventions from another repository.